core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -121,6 +123,12 @@ class HistoryServer(
   def initialize() {
     attachPage(new HistoryPage(this))
 
+    attachHandler(createRedirectHandler(
+      "/logs/delete",
+      "/",
+      handleDeleteLogRequest,
+      httpMethods = Set("GET", "POST")))
+
```

Comment:
it shouldn't be a GET for something side effecting; breaks the whole HTTP "no side effects on a GET" rule. Better: support DELETE for high-rest; POST if you want to support HTML <= 4.